source/adapters/hip/CMakeLists.txt

```diff
@@ -10,13 +10,13 @@ set(UR_HIP_PLATFORM "AMD" CACHE STRING "UR HIP platform, AMD or NVIDIA")
 
 # Set default ROCm installation directory
 set(UR_HIP_ROCM_DIR "/opt/rocm" CACHE STRING "ROCm installation dir")
-
-set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include")
-
-set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include")
-
-# Set HIP lib dir
-set(UR_HIP_LIB_DIR "${UR_HIP_ROCM_DIR}/lib")
+# Allow custom location of HIP/HSA include and HIP library directories
+set(UR_HIP_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include" CACHE PATH
+  "Custom ROCm HIP include dir")
+set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/hsa/include" CACHE PATH
```

Comment:
For `rocm-6.0.0`, the path to `hsa` directory has changed from `rocm/hsa/include` to `rocm/include/hsa`, so you may want to fix that here. This path mismatch has been addressed in the `intel/llvm` already, but I am getting `hsa directory not found` error if I don't apply the same workaround to this cmake file.

https://github.com/intel/llvm/blob/sycl/sycl/plugins/hip/CMakeLists.txt#L29

Reply:
I'm not sure how to best handle it, given the fact that `UR_HIP_HSA_INCLUDE_DIR` should be a cache variable. Maybe checking for the new location just before issuing an error would work?

```diff
diff --git a/source/adapters/hip/CMakeLists.txt b/source/adapters/hip/CMakeLists.txt
index 99c7cb23..e2a4e33a 100644
--- a/source/adapters/hip/CMakeLists.txt
+++ b/source/adapters/hip/CMakeLists.txt
@@ -30,10 +30,17 @@ if("${UR_HIP_PLATFORM}" STREQUAL "AMD")
                         " please check ROCm installation.")
     endif()

-    # Check if HSA include path exists
+    # Check if HSA include path exists. In rocm-6.0.0 the layout of HSA
+    # directory has changed, check for the new location as well.
     if(NOT EXISTS "${UR_HIP_HSA_INCLUDE_DIR}")
-    message(FATAL_ERROR "Couldn't find the HSA include directory at '${UR_HIP_HSA_INCLUDE_DIR}',"
-                        " please check ROCm installation.")
+      message(WARNING "Couldn't find the HSA include directory at "
+                      "'${UR_HIP_HSA_INCLUDE_DIR}'.")
+      set(UR_HIP_HSA_INCLUDE_DIR "${UR_HIP_ROCM_DIR}/include")
+      if(NOT EXISTS "${UR_HIP_HSA_INCLUDE_DIR}")
+        message(FATAL_ERROR "Couldn't find the HSA include directory at "
+                            "'${UR_HIP_HSA_INCLUDE_DIR}', please check ROCm "
+                            "installation.")
+      endif()
     endif()
 endif()


```

Reply:
Yeah, I think it should work.

Reply:
Great, that's done in 26299edb2cdcdab883a7b40a5d07a26ed847ce76
Thank you!